difference-of-squares/difference-of-squares.sml

```diff
@@ -0,0 +1,8 @@
+fun square_of_sum n =
+  raise Fail "'square_of_sum' is not implemented"
```

Comment:
I've looked at the other exercises in this track and it doesn't look like we've provided a failing stub for any of the other exercises.

For consistency I'd suggest dropping this file, and opening an issue in the repo to discuss whether or not it would be useful and `@`-mention anyone who has submitted exercises to get a general opinion.